jena-core/src/main/java/org/apache/jena/ext/xerces/impl/validation/ValidationState.java

```diff
@@ -169,16 +182,21 @@ public boolean isEntityUnparsed (String name) {
     // id
     @Override
     public boolean isIdDeclared(String name) {
+        if (fIdTable == null) {
+            return false;
+        }
         return fIdTable.containsKey(name);
     }
     @Override
     public void addId(String name) {
+        ensureIdTablesAreInitialized();
         fIdTable.put(name, fNullValue);
     }
 
     // idref
     @Override
     public void addIdRef(String name) {
+        ensureIdTablesAreInitialized();
```

Comment:
Only init idRefTableon demand here. 
fIdTable  is used in oder methods, so it might be that only one of the maps is needed.
```java
if(idRefTable== null) {
    idRefTable= new HashMap();
}
```